
# XYZ Company has offices in four different zones. The company wishes to investigate the following :
### ● The mean sales generated by each zone.
### ● Total sales generated by all the zones for each month.
### ● Check whether all the zones generate the same amount of sales.

In [1]:
import pandas as pd
import numpy as np

In [4]:
data=pd.read_csv('Sales_data_zone_wise.csv')

In [5]:
data


Month  Zone - A  Zone - B  Zone - C  Zone - D
0    Month - 1   1483525   1748451   1523308   2267260
1    Month - 2   1238428   1707421   2212113   1994341
2    Month - 3   1860771   2091194   1282374   1241600
3    Month - 4   1871571   1759617   2290580   2252681
4    Month - 5   1244922   1606010   1818334   1326062
5    Month - 6   1534390   1573128   1751825   2292044
6    Month - 7   1820196   1992031   1786826   1688055
7    Month - 8   1625696   1665534   2161754   2363315
8    Month - 9   1652644   1873402   1755290   1422059
9   Month - 10   1852450   1913059   1754314   1608387
10  Month - 11   2004480   1740365   1317869   1970069
11  Month - 12   1537539   1875658   1237722   1460165
12  Month - 13   1254939   1588473   1348629   1733383
13  Month - 14   1821799   1844081   1617376   1872259
14  Month - 15   1128185   1804613   1767047   1234311
15  Month - 16   1491534   1542053   1882159   1590913
16  Month - 17   1182110   1552388   2234469   2180416
17  Month - 18   1770324   1699213   1249821   2364132
18  Month - 19   1181635   1959268   2009168   1821882
19  Month - 20   1305972   1559670   2107949   2151008
20  Month - 21   1570152   2013615   1540016   2265814
21  Month - 22   1481619   1527574   2255729   2295079
22  Month - 23   1887694   1992155   1348387   1459683
23  Month - 24   1880820   1752873   2098463   2052591
24  Month - 25   1256333   1622671   1521792   1695122
25  Month - 26   1422853   1715465   1853636   1520406
26  Month - 27   1384426   1983163   1611169   1289160
27  Month - 28   1616640   1547991   2128022   2178267
28  Month - 29   1310654   1660092   1947119   1854412

In [6]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 29 entries, 0 to 28
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   Month     29 non-null     object
 1   Zone - A  29 non-null     int64 
 2   Zone - B  29 non-null     int64 
 3   Zone - C  29 non-null     int64 
 4   Zone - D  29 non-null     int64 
dtypes: int64(4), object(1)
memory usage: 1.3+ KB


## The mean sales generated by each zone

In [8]:
data .describe().round(2)

Zone - A    Zone - B    Zone - C    Zone - D
count       29.00       29.00       29.00       29.00
mean   1540493.14  1755559.59  1772871.03  1842926.76
std     261940.06   168389.89   333193.72   375016.48
min    1128185.00  1527574.00  1237722.00  1234311.00
25%    1305972.00  1606010.00  1523308.00  1520406.00
50%    1534390.00  1740365.00  1767047.00  1854412.00
75%    1820196.00  1875658.00  2098463.00  2180416.00
max    2004480.00  2091194.00  2290580.00  2364132.00

* ZONE-A Mean is 1540493.14
* ZONE-B Mean is 1755559.59
* ZONE-C Mean is 1772871.03
* ZONE-D Mean is 1842926.76

## Total sales generated by all the zones for each month.

In [9]:
data['Total sale for the month by all zones'] = data.sum(axis=1)

In [10]:
data

Month  Zone - A  Zone - B  Zone - C  Zone - D  \
0    Month - 1   1483525   1748451   1523308   2267260   
1    Month - 2   1238428   1707421   2212113   1994341   
2    Month - 3   1860771   2091194   1282374   1241600   
3    Month - 4   1871571   1759617   2290580   2252681   
4    Month - 5   1244922   1606010   1818334   1326062   
5    Month - 6   1534390   1573128   1751825   2292044   
6    Month - 7   1820196   1992031   1786826   1688055   
7    Month - 8   1625696   1665534   2161754   2363315   
8    Month - 9   1652644   1873402   1755290   1422059   
9   Month - 10   1852450   1913059   1754314   1608387   
10  Month - 11   2004480   1740365   1317869   1970069   
11  Month - 12   1537539   1875658   1237722   1460165   
12  Month - 13   1254939   1588473   1348629   1733383   
13  Month - 14   1821799   1844081   1617376   1872259   
14  Month - 15   1128185   1804613   1767047   1234311   
15  Month - 16   1491534   1542053   1882159   1590913   
16  Month - 17   1182110   1552388   2234469   2180416   
17  Month - 18   1770324   1699213   1249821   2364132   
18  Month - 19   1181635   1959268   2009168   1821882   
19  Month - 20   1305972   1559670   2107949   2151008   
20  Month - 21   1570152   2013615   1540016   2265814   
21  Month - 22   1481619   1527574   2255729   2295079   
22  Month - 23   1887694   1992155   1348387   1459683   
23  Month - 24   1880820   1752873   2098463   2052591   
24  Month - 25   1256333   1622671   1521792   1695122   
25  Month - 26   1422853   1715465   1853636   1520406   
26  Month - 27   1384426   1983163   1611169   1289160   
27  Month - 28   1616640   1547991   2128022   2178267   
28  Month - 29   1310654   1660092   1947119   1854412   

    Total sale for the month by all zones  
0                                 7022544  
1                                 7152303  
2                                 6475939  
3                                 8174449  
4                                 5995328  
5                                 7151387  
6                                 7287108  
7                                 7816299  
8                                 6703395  
9                                 7128210  
10                                7032783  
11                                6111084  
12                                5925424  
13                                7155515  
14                                5934156  
15                                6506659  
16                                7149383  
17                                7083490  
18                                6971953  
19                                7124599  
20                                7389597  
21                                7560001  
22                                6687919  
23                                7784747  
24                                6095918  
25                                6512360  
26                                6267918  
27                                7470920  
28                                6772277

* Total sale is showing in last column

## Check whether all the zones generate the same amount of sales.

#### Using One Way Annova
Using One-way ANOVA :
Level of significance, alpha = 0.05
* Null Hypothesis, H0 : All zones generate the same amount of sales
* Alternate Hypothesis, H1 : All zones generate different amount of sales

In [18]:
import scipy.stats as scp
from scipy.stats import f_oneway

In [19]:
fvalue,pvalue=scp.f_oneway(data['Zone - A'],data['Zone - B'],data['Zone - C'],data['Zone - D'])
fcriteria=scp.f.ppf(1-0.05,3,112)

In [20]:
fcriteria=scp.f.ppf(1-0.05,3,112)

In [21]:


print('F Statistics=',fvalue)
print('F criteria=',fcriteria)
print('pvalue=',pvalue)
print('significance level=0.05')



F Statistics= 5.672056106843581
F criteria= 2.6856434512366034
pvalue= 0.0011827601694503335
significance level=0.05


In [23]:
if fvalue>fcriteria and pvalue<0.05:
   print('Reject H0: All zones generate the same amount of sales')
else:
   print('Reject H1: All zones generate different amount of sales')

Reject H0: All zones generate the same amount of sales


* Sales are different is zones.